src/engine.ts

```diff
@@ -210,13 +211,47 @@ export function initializePolyfill() {
       scheduleUpdate();
     }
   });
+  const mutationObserver = createMutationObserver();
   mutationObserver.observe(documentElement, {
     childList: true,
     subtree: true,
     attributes: true,
     attributeOldValue: true,
   });
 
+  const originalAttachShadow = Element.prototype.attachShadow;
+  Element.prototype.attachShadow = function (options) {
+    const shadow = originalAttachShadow.apply(this, [options]);
+    getOrCreateInstance(shadow);
+    return shadow;
+  }
+
+  const originalReplaceSync = CSSStyleSheet.prototype.replaceSync;
+  if (originalReplaceSync) {
+    CSSStyleSheet.prototype.replaceSync = function (options) {
+      const result = transpileStyleSheet(
+        options,
+        undefined
+      );
+      setDescriptorsForStyleSheet(this, result.descriptors);
+      const replaceSync = originalReplaceSync.apply(this, [result.source]);
+      return replaceSync;
+    }
+  }
+
+  const originalReplace = CSSStyleSheet.prototype.replace;
```

Comment:
For completeness, I think we would also want to patch `insertRule` and `deleteRule`. For now though, I think you could just include a TODO comment about how they aren't handled yet.